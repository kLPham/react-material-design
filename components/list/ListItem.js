import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '@material/list/dist/mdc.list.css';
import '@material/drawer/dist/mdc.drawer.css';

// TODO: alt from avatarUrl
// TODO: When the list has an avatar but no leftAvatarUrl is provided, make sure that the text aligns with other li that do have an avatar.
// TODO: activeClassName isn't working.
const ListItem = ({ to, value, icon, leftAvatarUrl, rightIcon, rightTitle }) => {
    if (to) {
        return (<NavLink activeClassName="mdc-permanent-drawer--selected" className="mdc-list-item" to={to}>
            {icon && <i className="mdc-list-item__start-detail material-icons" aria-hidden>{icon}</i>}
            {leftAvatarUrl && <img className="mdc-list-item__start-detail" src={leftAvatarUrl} width="56" height="56" alt="" />}
            {value}
        </NavLink>);
    }
    return (<li className="mdc-list-item">
        {icon && <i className="mdc-list-item__start-detail material-icons" aria-hidden>{icon}</i>}
        {leftAvatarUrl && <img className="mdc-list-item__start-detail" src={leftAvatarUrl} width="56" height="56" alt="" />}
        {value}
        {rightIcon && <a className="mdc-list-item__end-detail material-icons" aria-label={rightTitle} title={rightTitle}>
            {rightIcon}
        </a>}
    </li>);
};

ListItem.propTypes = {
    icon: PropTypes.string,
    leftAvatarUrl: PropTypes.string,
    rightIcon: PropTypes.string,
    rightTitle: PropTypes.string,
    value: PropTypes.string.isRequired,
};

export default ListItem;
