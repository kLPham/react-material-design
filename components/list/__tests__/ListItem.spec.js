import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

describe('<ListItem />', () => {
    it('should render a basic list item', () => {
        const value = 'Single-line item';
        const wrapper = shallow(
            <ListItem value={value} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-item');
        expect(wrapper).toHaveText(value);
    });
    it('should render start (left) detail elements', () => {
        const value = 'Wi-Fi';
        const leftIcon = 'network_wifi';
        const wrapper = shallow(
            <ListItem value={value} icon={leftIcon} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-item');
        expect(wrapper.find('i')).toBePresent();
        expect(wrapper.find('i')).toHaveClassName('mdc-list-item__start-detail material-icons');
        expect(wrapper.find('i')).toHaveText(leftIcon);
    });
    it('should render start (left) detail avatar', () => {
        const value = 'George Washington';
        const leftAvatarUrl = 'http://4vector.com/i/free-vector-george-washington-clip-art_108422_George_Washington_clip_art_hight.png';
        const wrapper = shallow(
            <ListItem value={value} leftAvatarUrl={leftAvatarUrl} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-item');
        expect(wrapper.find('li')).toHaveText(value);
        expect(wrapper.find('img')).toBePresent();
        expect(wrapper.find('img')).toHaveClassName('mdc-list-item__start-detail');
        expect(wrapper.find('img')).toHaveProp('src');
    });
    it('should render end (right) detail elements', () => {
        const value = 'Janet Perkins';
        const rightIcon = 'favorite';
        const rightTitle = 'Remove from favorites';
        const wrapper = shallow(
            <ListItem value={value} rightIcon={rightIcon} rightTitle={rightTitle} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-item');
        expect(wrapper.find('a')).toBePresent();
        expect(wrapper.find('a')).toHaveClassName('mdc-list-item__end-detail material-icons');
        expect(wrapper.find('a')).toHaveText(rightIcon);
        expect(wrapper.find('a')).toHaveProp('title');
    });
});
