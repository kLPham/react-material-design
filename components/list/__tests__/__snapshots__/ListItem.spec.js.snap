exports[`<ListItem /> should render a basic list item 1`] = `
<li
  className="mdc-list-item">
  Single-line item
</li>
`;

exports[`<ListItem /> should render end (right) detail elements 1`] = `
<li
  className="mdc-list-item">
  Janet Perkins
  <a
    aria-label="Remove from favorites"
    className="mdc-list-item__end-detail material-icons"
    title="Remove from favorites">
    favorite
  </a>
</li>
`;

exports[`<ListItem /> should render start (left) detail avatar 1`] = `
<li
  className="mdc-list-item">
  <img
    className="mdc-list-item__start-detail"
    height="56"
    role="presentation"
    src="http://4vector.com/i/free-vector-george-washington-clip-art_108422_George_Washington_clip_art_hight.png"
    width="56" />
  George Washington
</li>
`;

exports[`<ListItem /> should render start (left) detail elements 1`] = `
<li
  className="mdc-list-item">
  <i
    aria-hidden={true}
    className="mdc-list-item__start-detail material-icons">
    network_wifi
  </i>
  Wi-Fi
</li>
`;
